// react-native-progress-dialog/src/ProgressDialog.js
import React, {Component} from "react";
import {
    View,
    Modal,
    ActivityIndicator,
    StyleSheet,
    Text
} from 'react-native';

class ProgressDialog extends Component {

    render() {

        const {visible} = this.props;
        const label = this.props.label ? this.props.label : 'Please wait...';
        const loaderColor = this.props.loaderColor ? this.props.loaderColor : '#0d0';
        const labelStyle = this.props.labelStyle ? this.props.labelStyle : {};

        return (
            <Modal
                animationType="none"
                transparent={true}
                visible={visible}
                onRequestClose={() => {

                }}>
                <View style={styles.wrapper}>
                    <View style={styles.content}>
                        <ActivityIndicator size="large" color={loaderColor}/>
                        <Text style={[styles.label, labelStyle]}>{label}</Text>
                    </View>
                </View>
            </Modal>
        );
    }
}

const styles = StyleSheet.create({
    label: {
        color: '#00000089',
        fontSize: 18,
        fontWeight: '700',
        textAlignVertical: 'center',
        marginLeft: 15

    },
    content: {
        flexDirection: 'row',
        backgroundColor: '#fff',
        height: 80,
        borderRadius: 5,
        width: '80%',
        padding: 20
    },
    wrapper: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'rgba(0,0,0,0.7)'
    }

});


export default ProgressDialog;
